### DISCUSSION TITLE
lib/data/items 的文献数据是怎么 format 的？

### DISCUSSION THREAD

**Discussion by zepinglee** [zotero-chinese/styles#409] [Q&A] [answered]
我在 Zotero 中更新了用于默认测试的文献的数据，但是导出到本仓库的 `lib/data/items` 时出现了过多的 diff。主要是日期的格式。不方便比对。

@northword `lib/data/items/*-data.json` 在迁移到这个位置时是怎么 format 的？

<img width="1033" alt="Screenshot 2024-11-22 at 21 54 06" src="https://github.com/user-attachments/assets/5b0cbf4c-75a4-43d6-8b1e-b05314993222">

**Comment by northword** [comment, marked as answer]
这里的 format 是由 prettier 控制的，配置文件在 .prettierrc。

可能是是 printWidth 达到了限制强制换了行，也可能是 prettier 对 array 类型的格式化确实是这样导致的。

我这会儿不在电脑前，明天调查这个原因。

---

edited

似乎是 printwidth 的原因，也许需要把这里调大，比如 120：

https://github.com/zotero-chinese/styles/blob/ba80cd1db70e466757145d6bb58f39e0791658a3/.prettierrc#L3

**Reply by zepinglee** [reply]
感谢！

**Comment by northword** [comment]
我试了下，在 VS Code 里使用 “JSON 语言功能” 格式化 JSON 文件时，会生成下面的代码：

```json
	{
		"id": "ssc.2.6.2.5:01",
		"type": "personal_communication",
		"archive": "National Archives II, College Park, MD",
		"archive_location": "Box 1032, NSC Files, Nixon Presidential Material Project (NPMP)",
		"language": "en-US",
		"title": "Nixon to Kissinger",
		"issued": {
			"date-parts": [
				[
					"1969",
					2,
					1
				]
			]
		}
	},
```

当使用 Prettier 格式化时，会出现下面的：

```json
	{
		"id": "ssc.2.6.2.5:01",
		"type": "personal_communication",
		"archive": "National Archives II, College Park, MD",
		"archive_location": "Box 1032, NSC Files, Nixon Presidential Material Project (NPMP)",
		"language": "en-US",
		"title": "Nixon to Kissinger",
		"issued": {
			"date-parts": [["1969", 2, 1]]
		}
	},
```

也许你需要在 VS Code 里安装 Prettier 插件，或者，可以始终使用 JSON 语言功能作为默认格式化器，并移除 Prettier 相关的依赖项。
